### ISSUE TITLE
Images added via collection update API not showing in app

### ISSUE THREAD

**Issue by NecoHorne** [directus/directus#16043]
### Describe the Bug

I use the file upload API `POST /files/import` to upload an image file and grab the id from the response and add it to the image field of my collection and update via the update API `PATCH /items/:collection/:id`

The images do not show in the app, I have checked to make sure that they are in the correct folder via the upload and they are. When checking the database under the collection the ID for the image is in the correct column and the image does exist in the `directus_files` table. Checked this locally and on my deployed instance. 

![image](https://user-images.githubusercontent.com/30078491/196452677-353a46b7-744e-429b-8f38-e83147a0f6ea.png)

also shows nothing in the item
![image](https://user-images.githubusercontent.com/30078491/196455758-4bcb6c83-aaa3-46dc-adb7-7b6bee6dbba7.png)

But it does show the file id in the database

### To Reproduce

1. Upload image via file import API `POST /files/import`
2. Grab the file id from the response and update an image field n a collection via the items update API `PATCH /items/:collection/:id`

### Errors Shown

No Errors shown in browser tools or directus logs on the vm.

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

16.15.0

### What database are you using?

MySQL

### What browser are you using?

Chrome

### How are you deploying Directus?

Ubuntu EC2 instance

**Comment by br41nslug** [comment]
Do the images show up in the Directus file library? (if so it may be this merged PR https://github.com/directus/directus/pull/15736 will be included in the next release)
That second screenshot makes me think however that something is going wrong with the updating of the item as that looks like the interfaces are showing the same as it would with a `NULL` value. Can you check/share the PATCH request?

**Comment by NecoHorne** [comment]
> 

Yes they are in the Directus file library, I can see them there. I checked the folder in my upload request to make sure its also the same as the one I set in the data model for the images location. Also double checked to make sure that the image id in the collection corresponds to the correct image id in the `directus_files` table. If I do a `GET` request to the `assets/file_id` or `assets/file_location_disk` there is no issue. Just on the app in the item collection where they don't show. 

I will wait for the next release, maybe that update solves the issue.

**Comment by ConnorSimply** [comment]
- When you perform a GET request on the item is the image ID there?
- Are you viewing from an admin user or someone with permissions?

**Comment by NecoHorne** [comment]
As Admin user. 

Yes when I perform a get the ID for the image is there but, only the ID when I use the fields operator with `*.*.*` it still only shows the id and not the Directus file object like it usually would.

**Comment by ConnorSimply** [comment]
That sounds like the relationship isn't set up correctly, do you mind sharing the field config for the image field?

**Comment by NecoHorne** [comment]
@ConnorSimply well spotted. I pulled the DDL from the DB and saw that the foreign key constraint was not set up. What is interesting is I added this field via the app, in the app everything looked good but the relationship was not set up.  
![image](https://user-images.githubusercontent.com/30078491/196614201-925427e9-d7c1-411a-b033-58830a620b62.png)

I added the constraint manually and now it shows in the edit section. Still not in the list view but I am assuming that is the issue @br41nslug mentioned before.
![image](https://user-images.githubusercontent.com/30078491/196614409-fa682e5a-2702-41d5-b562-2b20b9220e92.png)
![image](https://user-images.githubusercontent.com/30078491/196614561-97ef08c0-be0a-4629-875c-213a958ff08c.png)

Thanks for the help. Just for some extra info I added that field and another image field in another item after updating to v9.18.1, both had the same issue, not sure if its a bug in v9.18.1 or something else just went wrong there.
